scipy/optimize/_optimize.py

```diff
@@ -3358,8 +3358,11 @@ def brute(func, ranges, args=(), Ns=20, full_output=0, finish=fmin,
         `multiprocessing.Pool.map` for evaluating the grid in parallel.
         This evaluation is carried out as ``workers(func, iterable)``.
         Requires that `func` be pickleable.
-
-        .. versionadded:: 1.3.0
+    integrality : 1-D array, optional
+        For each decision variable, a boolean value indicating whether the
```

Comment:
In note 2 the user is told what the size of the output grid is supposed to be. Note 2 needs to be amended to explain how the grid size is amended when integrality is used for a given parameter. i.e. the dimensionality can be reduced significantly.

Reply:
Furthermore, if a user supplies a `slice` object in ranges they need to be told how those slices are interpreted in an integrality context.

Reply:
Agreed. I think I should set `integrality` to `False` wherever the user provides a slice and document that.

Reply:
If integrality can be implemented by using a slice for a parameter: `slice(round(low), round(high), 1)`, then is it ultimately necessary to add integrality to `brute`?

Reply:
For convenience and clarity, I think yes. A long time ago I tried using `slice`s to effectively enforce integrality constraints in `brute` from the outside and it was just messy. The goal here is to wrap `brute` in this iterative refinement function, so it will be a lot cleaner if `brute` just has `integrality`.

Reply:
> Agreed. I think I should set integrality to False wherever the user provides a slice and document that. The only thing we might want it for in that case is to fix the integral variables during polishing.

At the moment we've got `integrality` overriding `slice`. That's fine by me, but note 2 needs to be updated to say how the grid is changed w.r.t integrality